src/daria/analysis/segmentationcomparison.py

```diff
@@ -1,113 +1,471 @@
+                f" in {self.segmentation_names[i]}"
+            ] = self.colors[i, 1]
+
+        for i in range(self.number_of_segmented_images - 1):
+            self.color_dictionary[
+                f"{i+2} segmentations overlaps in {self.component_names[0]}"
+            ] = self.gray_base[0] * (i + 1)
+            self.color_dictionary[
+                f"{i+2} segmentations overlaps in {self.component_names[1]}"
+            ] = self.gray_base[1] * (i + 1)
+            self.color_dictionary[
+                f"Mixture {self.component_names[0]} and "
+                f"{self.component_names[1]} for {i+2} segmentations"
+            ] = self.gray_base[2] * (i + 1)
+
+    def __call__(
+        self,
+        *segmentations,
+        plot_result: bool = False,
+        **kwargs,
+    ) -> np.ndarray:
+        """
+        Comparison of segmentations.
+
+        Args:
+            segmentations (asterisk argument): Allows to provide an arbitraty
+                number of segmented numpy arrays or da.Images of integers to be compared
+            Optional keyword arguments (kwargs):
+                plot_result (bool): plots the result with matplotlib if True,
+                    default is False.
+                roi (tuple): roi where the segmentations should be compared,
+                    if default the maximal roi that fits in all segmentations
+                    is chosen.
+
+        """
+
+        # Define number of segmentations
+        assert self.number_of_segmented_images == len(segmentations)
+
+        if "roi" in kwargs:
+            roi = kwargs["roi"]
+            return_image: np.ndarray = np.zeros(
+                (roi[0].stop - roi[0].start, roi[1].stop - roi[1].start) + (3,),
+                dtype=np.uint8,
+            )
+
+        else:
+            if isinstance(segmentations[0], np.ndarray):
+                nx = min([seg.shape[0] for seg in segmentations])
+                ny = min([seg.shape[1] for seg in segmentations])
+            elif isinstance(segmentations[0], da.Image):
+                nx = min([seg.img.shape[0] for seg in segmentations])
+                ny = min([seg.img.shape[1] for seg in segmentations])
+            roi = (slice(0, nx), slice(0, ny))
+            return_image: np.ndarray = np.zeros((nx, ny) + (3,), dtype=np.uint8)
+
+        # Enter gray everywhere there are ovelaps of different segmentations
+
+        if isinstance(segmentations[0], np.ndarray):
+            for k in range(self.number_of_segmented_images):
+                for i in range(k + 1, self.number_of_segmented_images):
+
+                    # Overlap of components
+                    for c_num, c in enumerate(self.components):
+                        return_image[
+                            np.logical_and(
+                                segmentations[k][roi] == c,
+                                segmentations[i][roi] == c,
+                            )
+                        ] += self.gray_base[c_num]
```

Comment:
```suggestion
                        ] = self.gray_base[c_num]
```
To be consistent with the last assignment.

Reply:
same?